### PR TITLE
Expose the subiquity API to set the variant ('server'/'desktop'), and use it.

### PR DESCRIPTION
**Pull request by oSoMoN**

### CODE REVIEW

File: packages/subiquity_client/lib/src/types.dart [hunk 1]
```diff
@@ -6,6 +6,8 @@ part 'types.g.dart';
 
 // ignore_for_file: invalid_annotation_target
 
+enum Variant { SERVER, DESKTOP }
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:9]: was there one for WSL?

Comment (oSoMoN) [reply packages/subiquity_client/lib/src/types.dart:9]: No, only "server" and "desktop": https://github.com/canonical/subiquity/blob/main/subiquity/server/server.py#L240

Comment (oSoMoN) [reply packages/subiquity_client/lib/src/types.dart:9]: We might need/want a WSL variant added to subiquity though, I guess this needs to be discussed with @dbungert and @jibel.

**Review by jpnurmi — APPROVED** [review]
Ok, LGTM :+1:

File: packages/subiquity_client/lib/src/types.dart (hunk 1 above)
Comment (dbungert) [reply packages/subiquity_client/lib/src/types.dart:9]: yes, there are 2 wsl variant declarations, on one of the WSL branches.  Also a get() function for getting the variant.

Comment (oSoMoN) [reply packages/subiquity_client/lib/src/types.dart:9]: Ack, thanks for the info. I subscribed to https://github.com/canonical/subiquity/pull/1042, and once it lands and we update our subiquity checkout I'll add the "wsl_setup" and "wsl_configuration" variants as well as a getter.

**Pull request merged**